### PR TITLE
Date tests

### PR DESCRIPTION
**Pull request by flippinjoe**
- Updated test class names causing some weird test issues
- Added URBNDateRule class
- Added URBNDateRule tests

### CODE REVIEW

File: Pod/Classes/URBNValidationRules.swift [hunk 1]
```diff
@@ -94,3 +54,36 @@ public class URBNBlockRule: URBNBaseRule {
...
+        case .Past: return NSComparisonResult.OrderedDescending
+        case .Future: return NSComparisonResult.OrderedAscending
+        }
+    }
+}
+
+public class URBNDateRule: URBNBaseRule {
+    public var comparisonType = URBNDateComparision.Past
+    
+    public init(comparisonType: URBNDateComparision = URBNDateComparision.Past) {
```
Comment (ndis1) [inline Pod/Classes/URBNValidationRules.swift]: `URBNDateComparision` can be inferred

File: Pod/Classes/URBNBaseRule.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+        set {
+            _localizationKey = newValue
+        }
+    }
+    
+    public init(localizationKey: String? = nil) {
+        if localizationKey != nil && localizationKey?.length > 0 {
+            self.localizationKey = localizationKey!
+        }
```
Comment (ndis1) [inline Pod/Classes/URBNBaseRule.swift:41]: I'd rather do this as 

```
if let key = localizationKey where key.length > 0 {
    self.localizationKey = key
}
```

File: Pod/Classes/URBNValidationRules.swift [hunk 3]
```diff
@@ -94,3 +54,36 @@ public class URBNBlockRule: URBNBaseRule {
...
+        self.comparisonType = comparisonType
+        super.init()
+    }
+    
+    public override func validateValue<T: NSDate>(value: T?) -> Bool {
+        if value == nil { return false }
+        
+        return NSDate().compare(value!) == comparisonType.comparisonResult
```
Comment (ndis1) [inline Pod/Classes/URBNValidationRules.swift]: this one is 6 to one half-dozen to the other, but I like guard for these guys

File: Pod/Classes/URBNBaseRule.swift (hunk 2 above)
Comment (jgrandelli) [inline Pod/Classes/URBNBaseRule.swift:41]: can't you `isEmpty` here?

File: Pod/Classes/URBNValidationRules.swift (hunk 1 above)
Comment (flippinjoe) [inline Pod/Classes/URBNValidationRules.swift]: Fixed

File: Pod/Classes/URBNBaseRule.swift (hunk 2 above)
Comment (flippinjoe) [inline Pod/Classes/URBNBaseRule.swift:41]: Can't do isEmpty on nil

Comment (flippinjoe) [inline Pod/Classes/URBNBaseRule.swift:41]: Could do `localizationKey != nil && !localizationKey!.isEmpty`

**Pull request merged**